File: php/examples/patch_payment_methods.php
<?php
require_once 'base_request.php';

// Example of PATCH /payment-methods/:id

$debug = true;
$payment_method_id = $_ENV['PAYMENT_METHOD_ID'];
$card_number = '2223003122003222';
$exp_month = '10';		
$exp_year = '2031';					
$cvc = '368';
$first_name = 'Oscar';
$last_name = 'Grouch';
$billing_address = array(
    "city" =>  "New York",
    "state" =>  "NY",
    "zip" =>  "36401",
    "country" =>  "US",
    "address1" =>  "5643 Seasame Street",
    "address2" =>  ""
  );

$request_data = array(
	'number' => $card_number,
	'exp_month' =>  $exp_month,
	'exp_year' =>  $exp_year,
	'cvc' => $cvc,
	'first_name' => $first_name,
	'last_name' =>  $last_name,
	'billing_address' =>  $billing_address
);

$request_params = null;

$json_body = json_encode($request_data);

send_request('PATCH', "/payment-methods/$payment_method_id", $request_params, $json_body);

File: csharp/examples/PatchPaymentMethods.cs
using System;
using Newtonsoft.Json;

namespace Examples.examples
{
    // Example PATCH /payment-methods/:payment_method_id

    public class PatchPaymentMethods : BaseRequest
    {
        private string _paymentMethodId;
        public PatchPaymentMethods()
        {
            dotenv.net.DotEnv.Load();
            _paymentMethodId = Environment.GetEnvironmentVariable("PAYMENT_METHOD_ID");
            Console.WriteLine($"PatchPaymentMethods: _paymentMethodId={_paymentMethodId}");
        }

        public override void SendRequest()
        {
            var billing_address = new BillingAddress
            {
                city = "New York",
                state = "NY",
                zip = "10463",
                country = "US",
                address1 = "20211 Seasame Street",
                address2 = ""
            };

            var payment_method = new PaymentMethod
            {
                number = "4000056655665556",
                exp_month = "03",
                exp_year = "2029",
                cvc = "543",
                first_name = "Elmo",
                last_name = "Grover",
                billing_address = billing_address
            };

            var body = JsonConvert.SerializeObject(payment_method, Formatting.None);

            PatchRequest($"payment-methods/{_paymentMethodId}", body);
        }
    }
}

